[CCI] Create index lifecycle guide

Nicksqain opened the pull request:
### Description

- [x] Create index lifecycle guide

Using the doc: [Opensearch rust](https://docs.rs/opensearch/latest/opensearch/)
### Issues Resolved
- closes #139 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

Xtansia reviewed (changes requested):
There are several code errors in the examples, I think it would be beneficial for you to create an accompanying `opensearch/examples/index_lifecycle.rs` example file that combines all the snippets together to ensure the code is valid, compiles and can be run successfully.

Nicksqain commented:
@Xtansia 
I apologise for making the push with the wrong code. The other PR had the same problem, but I added the source code there because the git pre_commit hook was not working correctly. I have now added the correct file and added a companion file to check.

Nicksqain commented:
@Xtansia I may have misunderstood, but was it necessary to include an accompanying .rs file in the PR?

dblock commented:
> @Xtansia I may have misunderstood, but was it necessary to include an accompanying .rs file in the PR?

I like it with the PR. I wish we had a way to just include the file in the guides instead of copy-pasting it, and/or at least running tests on the .rs code. Thinking out loud.

Xtansia commented on guides/index_lifecycle.md:
````diff
@@ -0,0 +1,164 @@
+
+The response body contains the index's settings and mappings:
+
+```rust
````
```suggestion
```json
```

Xtansia commented on guides/index_lifecycle.md:
````diff
@@ -0,0 +1,164 @@
+    .send().await?;
+```
+
+_Note_: The error_trace method allows for error tracing in the server response. If a 404 error occurs when removing the index, it will be ignored and the send method will return a successful result.
````
This is not accurate as `error_trace` does not cause errors to be ignored.

Nicksqain replied:
@Xtansia
```rust
 ignore_unavailable: Option<bool>
 ```
I found this in types, please check if it should ignore no index?

Xtansia replied:
Correct, `ignore_unavailable` will ignore any indices that do not exist

Xtansia replied:
The note here needs to be updated to be about the `ignore_unavailable` rather than about `error_trace`

Xtansia commented on opensearch/examples/index_lifecycle.rs:
```diff
@@ -0,0 +1,81 @@
+use opensearch::{ http::transport::TransportBuilder, OpenSearch, http::transport::SingleNodeConnectionPool, cert::CertificateValidation};
```
Please run `cargo fmt` to fix the formatting of this file.

Nicksqain commented on guides/index_lifecycle.md:
````diff
@@ -0,0 +1,164 @@
+docker-compose up -d
+```
+
+Let's create a client instance to access this cluster:
+
+```rust
+use opensearch::{ http::transport::TransportBuilder, OpenSearch, http::transport::SingleNodeConnectionPool, cert::CertificateValidation};
+use url::Url;
+use opensearch::{IndexParts, indices::{ IndicesCreateParts, IndicesExistsParts, IndicesPutSettingsParts, IndicesPutMappingParts, IndicesGetParts,IndicesDeleteParts}};
+use serde_json::json;
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    // Create a client to make API calls to OpenSearch running on http://localhost:9200.
+    let client = OpenSearch::default();
+
+    // Alternatively, you can create a client to make API calls against OpenSearch running on a specific url::Url.
+    let url = Url::parse("https://example.com")?;
+    let transport = TransportBuilder::new(SingleNodeConnectionPool::new(url)).cert_validation(CertificateValidation::None).build()?;
+    let client = OpenSearch::new(transport);
+    println!("{:?}", client.info().send().await?); // Check server info and make sure the client is connected
+    Ok(())
+}
+```
````
Thank you for reply!
Would that be wrong too? Was the error due to an incorrect URL?
```rust
// Create a client to make API calls to OpenSearch running on http://localhost:9200.
    let client = OpenSearch::new(Transport::single_node("http://localhost:9200")?);
    // Alternatively, you can create a client to make API calls against OpenSearch running on a specific url::Url.
    let url = Url::parse("https://localhost:9200")?;
    let credentials = Credentials::Basic("admin".into(), "admin".into());
    let transport = TransportBuilder::new(SingleNodeConnectionPool::new(url))
        .cert_validation(CertificateValidation::None)
        .auth(credentials)
        .build()?;
    let client = OpenSearch::new(transport);
```

Xtansia replied:
@Nicksqain I think as the idea of the guide is we're specifically connecting to the cluster run via `docker-compose` you only need to retain:
```rust
let url = Url::parse("https://localhost:9200")?;
let credentials = Credentials::Basic("admin".into(), "admin".into());
let transport = TransportBuilder::new(SingleNodeConnectionPool::new(url))
        .cert_validation(CertificateValidation::None)
        .auth(credentials)
        .build()?;
let client = OpenSearch::new(transport);
```

Xtansia commented on guides/index_lifecycle.md:
```diff
@@ -0,0 +1,174 @@
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    // Create a client to make API calls to OpenSearch running on http://localhost:9200.
```
```suggestion
    // Create a client to make API calls to OpenSearch running on https://localhost:9200.
```

Xtansia commented on opensearch/examples/index_lifecycle.rs:
```diff
@@ -0,0 +1,141 @@
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    // Create a client to make API calls to OpenSearch running on http://localhost:9200.
```
```suggestion
    // Create a client to make API calls to OpenSearch running on https://localhost:9200.
```

Xtansia reviewed (changes requested):
Looks good, just the last references to `error_trace` need to be updated

dblock commented:
Can we please link these guides from existing .mds?

Xtansia commented on guides/index_lifecycle.md:
````diff
@@ -0,0 +1,174 @@
+    .send().await?;
+```
+
+_Note_: The ignore_unavailable method allows for error tracing in the server response. If the index does not exist when removing the index, it will be ignored and the send method will return a successful result.
````
This note still references "error tracing" which is not what the `ignore_unavailable` parameter does
